casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala

```diff
@@ -163,20 +163,17 @@ object InterpreterUtil {
       runtimeManager: RuntimeManager[F],
       blockData: BlockData,
       invalidBlocks: Map[BlockHash, Validator]
-  ): F[(StateHash, StateHash, Seq[InternalProcessedDeploy])] =
+  ): F[(StateHash, StateHash, Seq[InternalProcessedDeploy])] = {
+    import shapeless.syntax.std.tuple._
     for {
       nonEmptyParents <- parents.pure
                           .ensure(new IllegalArgumentException("Parents must not be empty"))(
                             _.nonEmpty
                           )
       preStateHash <- computeParentsPostState(nonEmptyParents, dag, runtimeManager)
-      result <- runtimeManager
-                 .computeState(preStateHash)(deploys, blockData, invalidBlocks)
-                 .map {
-                   case (postStateHash, processedDeploys) =>
-                     (preStateHash, postStateHash, processedDeploys)
-                 }
-    } yield result
+      result       <- runtimeManager.computeState(preStateHash)(deploys, blockData, invalidBlocks)
+    } yield preStateHash +: result
```

Comment:
I know we have discussed it - but I have to say once again that this is really nice!